elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,283 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1712[ELY-1712]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-699[EAP7-699]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kašík]
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Support for RFC 5424 should be enhanced, according to the following parameters, all of which should be added to the ```syslogAuditLogType``` in the Elytron schema
+// Could not find information about this in legacy
+** A ```syslog-format``` attribute should be added which supports a value of ```RFC5424```
+** Some of the parameters needed for RFC 5424 already exist in the Elytron codebase, which are the following:
+*** severity
+*** server address
+*** port
+*** setting TCP
+*** setting SSL
+*** setting socket factory
+*** setting hostname
+** The current attributes available in the Elytron schema, which will be used for RFC 5424, are:
+*** server-address
+*** port
+*** transport type
+*** message format
+*** hostname (?)
+*** ssl-context
+** Additional attributes which will need to be added are:
+*** include timeQuality
+*** include origin
+*** include meta
+*** facility values
+*** priority mapping from facility and severity values, in accordance with https://tools.ietf.org/html/rfc5424#page-9[RFC 5424 Section 6.2.1]
+** The following attributes will also be available, but should be discarded if timeQuality is not specified:
+*** tzKnown
+*** isSynced
+*** syncAccuracy
+** The following attributes will also be available, but should be discarded if origin is not specified:
+*** ip
+*** enterpriseId
+*** software
+*** swVersion
+** The following attributes will also be available, but should be discarded if meta is not specified:
+*** sequenceId
+*** sysUpTime
+*** language
+
+* Support for RFC 3164 should be added, based off of the implementation for RFC 5424 above, as such should also be added to the ```syslogAuditLogType``` in the Elytron schema
+** The ```syslog-format``` attribute described in the RFC 5424 section above should also support a value of ```RFC3164```
+** All parts of RFC 3164 are included in the RFC 5424 specification, as such the support for RFC 3164 should be only
+changing the message format and not require any additional implementation
+** The implementation of the RFC 3164 message format can be implemented based on the https://github.com/kabir/tamper-detecting-audit-log/blob/master/src/main/java/org/jboss/audit/log/TcpSyslogHandler.java[tamper-detecting-audit-log]
+
+* Support for user formatted strings should be added
+** Currently, this feature does not have a way to be implemented
+
+* There should be the ability to specify a Java class should be logged by using an @Audit Java annotation, similarly to what was available in Picketbox
```

Comment:
Can you please elaborate on this? This analysis will be also used as a resource for documentation and it is not clear for me how the `@Audit` annotation should be used.

Reply:
This is one of the feature parity with legacy auditing parts. As far as I was able to tell, in PicketBox, the @Audit annotation can be added onto a class to signify that instances of that class being called should be logged. For the resource in Picketbox, see: https://developer.jboss.org/wiki/PicketBoxSecurityAnnotations#jive_content_id_SecurityAudit_Annotation

Reply:
Thank you. I know, it seems clear to you and it might seemingly be clear to me, but this analysis must be comprehensible for a new comer. For someone who have never used features provided in legacy audit log configuration. For example, it is possible, that technical writers who will be processing this document will not know anything about it, because they have no experience.

Reply:
I'll add the details from this comment to the proposal for it is better clarified